### PR TITLE
Facade schemas for 2.9.17

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following builds the client facades for 2.9.17. Unfortunately, the
schema generator doesn't currently work with 3.8. The fix is that kind
Any shouldn't be exploded when getting the type.

The clients are just updated with the latest changes.

### CODE REVIEW

File: juju/client/_client1.py [hunk 1]
```diff
@@ -2371,11 +2460,12 @@ class CAASFirewallerEmbeddedFacade(Type):
                                                                         'type': 'array'}},
                                              'required': ['results'],
                                              'type': 'object'}},
-     'properties': {'ApplicationCharmURLs': {'description': 'ApplicationCharmURLs '
```
Comment (SimonRichardson) [inline juju/client/_client1.py:2374]: This is scary that it was around and that it isn't any longer?

File: juju/client/_client1.py [hunk 2]
```diff
@@ -2732,6 +3049,29 @@ async def ApplicationsConfig(self, entities=None):
...
+
+        url : str
+        Returns -> Charm
+        '''
+        if url is not None and not isinstance(url, (bytes, str)):
+            raise Exception("Expected url to be a str, received: {}".format(type(url)))
+
+        # map input types to rpc msg
+        _params = dict()
+        msg = dict(type='CAASFirewaller',
```
Comment (SimonRichardson) [inline juju/client/_client1.py:3065]: What's the CAASFirewaller doing here?

File: juju/client/facade.py [hunk 3]
```diff
@@ -256,6 +256,8 @@ def __init__(self, schema, defs):
                     self.append((name, rtype))
 
     def do_explode(self, kind):
+        if kind is Any:
+            return False
```
Comment (SimonRichardson) [inline juju/client/facade.py:260]: This is the fix for generating the facades.

**Review by cderici — APPROVED** [review]

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**